src/main/java/org/fungover/haze/HazeDatabase.java

```diff
@@ -143,4 +144,20 @@ public void addValue(String key, String value) {
             lock.unlock();
         }
     }
+
+
+    public String getIndex(List<String> inputList){
+        if (inputList.size() != 2)
+            return ARGUMENT_ERROR;
+        String key = inputList.get(1);
+        lock.lock();
+        List<String> keys = new ArrayList<>(database.values());
+        try{
+            if(keys.contains(key)){
+                return "$" + keys.indexOf(key) + "\r\n";
```

Comment:
Reply should be:
[Bulk string reply](https://redis.io/docs/reference/protocol-spec#bulk-strings): the requested element.
[Nil reply](https://redis.io/docs/reference/protocol-spec#bulk-strings): when index is out of range.

Observe that according to the documentation: https://redis.io/commands/lindex/
 Negative indices can be used to designate elements starting at the tail of the list. Here, -1 means the last element, -2 means the penultimate and so forth.